apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/KotlinCodeGen.kt

```diff
@@ -119,11 +121,13 @@ internal class KotlinCodeGen(
         .filter { !context.resolver.canResolveSchemaType(it.name) }
         .forEach { iface ->
           builders.add(InterfaceBuilder(context, iface, generateDataBuilders))
+          builders.add(InterfacePossibleTypeEnumBuilder(context, iface))
```

Comment:
I would have made the enum a nested class of the main interface to not clutter the namespace too much:

```kotlin
public class Friend {
  public enum class FriendPossibleTypes(
     public val rawValue: String,
  } { /* ... */ }


  public companion object {
    public val type: UnionType = UnionType("Friend", Human.type, Human2.type)
  }
}
```